component/namespace.jsonnet

```
@@ -7,9 +7,13 @@ local params = inv.parameters.argocd;
 local ns_metadata = {
   metadata+: {
     labels+: {
-      'app.kubernetes.io/part-of': 'argocd',
-      'openshift.io/cluster-monitoring': 'true',
-    },
+               'app.kubernetes.io/part-of': 'argocd',
+               'openshift.io/cluster-monitoring': 'true',
+             } +
+             if std.member(inv.applications, 'networkpolicy') then {
+               [inv.parameters.networkpolicy.labels.noDefaults]: 'true',
+               [inv.parameters.networkpolicy.labels.purgeDefaults]: 'true',
+             } else {},
```

We should probably ensure that the labels are only added when parameter `network_policies.enabled=true` (otherwise we're actively making the configuration worse for users that use the default configuration of this component on a cluster which has component `networkpolicy`):

```suggestion
             if params.network_policies.enabled && std.member(inv.applications, 'networkpolicy') then {
               [inv.parameters.networkpolicy.labels.noDefaults]: 'true',
               [inv.parameters.networkpolicy.labels.purgeDefaults]: 'true',
             } else {},
```